src/feature-interface.h

```diff
@@ -0,0 +1,41 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <map>
+#include <memory>
+
+
+namespace librealsense {
+
+
+// Base class for all features
+class feature_interface
+{
+public:
+    feature_interface( const std::string& id ) : _id( id )
+    {
+    }
+
+    const std::string & get_id() const { return _id; }
+
+    virtual ~feature_interface() = default;
+
+private:
+    std::string _id;
+};
+
+typedef std::map< std::string, std::shared_ptr< feature_interface > > features;
```

Comment:
The more I deal with these kinds of typedefs, the more I dislike them: they require that we include a header to use.

I suggest we remove.

Reply:
Any specific reason why? It is more readable and can more easily be changed, say to a vector

Removed.